### PR TITLE
fix: allow redirect to external url

### LINKED ISSUE #770: Allow redirect to absolute URL
Feature request: Allow redirects to absolute URLs, SSR and client-side.

We have an app (https://domainr.com) with several legacy URLs, some of which redirect to other websites. We’re exploring using Nuxt/Vue for our new website and would like to preserve this behavior.

## Snippet to reproduce

`~/pages/_domain/iana.vue`

```javascript
<script>
export default {
  fetch ({ params, redirect }) {
    redirect(301, `https://www.iana.org/domains/root/db/${params.domain}.html`)
  }
}
</script>
```

## Steps to reproduce

Navigate to `/com/iana`

## Expected

Redirect to `https://www.iana.org/domains/root/db/com.html`

(Click here to see expected behavior: https://domainr.com/com/iana)

## What actually happens

Redirect to `/https:/www.iana.org/domains/root/db/com.html` (local URL)

## Background

Nuxt only allows local redirects to a path, which is prepended with `router.base` here: https://github.com/nuxt/nuxt.js/blob/dev/lib/app/server.js#L39

### PR DESCRIPTION
**Pull request by clarkdo**
resolves #770

### CODE REVIEW

**Review by pi0 — APPROVED** [review]
LTGM. Did you checked how does importing `url` impacts on production bundle size?

**Comment by clarkdo** [comment]
@pi0 
About 2.7kb(gzipped)

File: test/basic.csr.test.js [hunk 1]
```diff
@@ -143,6 +143,14 @@ test('/redirect2', async t => {
   t.is(await page.$text('h1'), 'Index page')
 })
 
+test('/redirect3', async t => {
+  // New page for redirecting to external link.
+  const page = await browser.page(url('/'))
+  await page.nuxt.navigate('/redirect3', false)
+  await page.waitForFunction(() => window.location.href === 'https://nuxtjs.org/')
+  t.pass()
```
Comment (Atinux) [inline test/basic.csr.test.js]: Don't forget to use `page.close()` here since you create a new instance :)

**Review by atinux — CHANGES_REQUESTED** [review]
Thanks for the PR @clarkdo :)

2.7kb is huge for only a redirect to a full url :/

Also, I'm not sure that we want to display an error page for this but keep instead the actual behavior to redirect directly on server-side.

File: test/basic.csr.test.js (hunk 1 above)
Comment (clarkdo) [reply test/basic.csr.test.js]: I'm trying to replace `url` and will change the test.
About the `ctx.app.error`, I didn't find an appropriate way to jump out of the rendering except `ctx.app.error` 😓

File: lib/app/utils.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import Vue from 'vue'
+import url from 'urlite/querystring/urlite'
```
Comment (pi0) [inline lib/app/utils.js]: Can we use `import { parse, format }` for even more optimizations too? :)

Comment (clarkdo) [reply lib/app/utils.js]: `'urlite/querystring/urlite'` only exports `parse` and `format`, but for avoiding future changing, I will make the change as your advice.

**Pull request closed without merging**